alkira/resource_alkira_peering_gateway_cxp.go

```diff
@@ -0,0 +1,193 @@
+package alkira
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/alkiranet/alkira-client-go/alkira"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAlkiraPeeringGatewayCxp() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Manage CXP Peering Gateways.",
+		CreateContext: resourceAlkiraPeeringGatewayCxpCreate,
+		ReadContext:   resourceAlkiraPeeringGatewayCxpRead,
+		UpdateContext: resourceAlkiraPeeringGatewayCxpUpdate,
+		DeleteContext: resourceAlkiraPeeringGatewayCxpDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the Peering Gateway." +
+					"Once deployed, this property can not be changed.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Description: "Description of the CXP Peering Gateway.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"cxp": {
+				Description: "The CXP to which the Gateway is attached." +
+					"Once deployed, this property can not be changed.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			// TODO: change cloud_provider to be a Required value when more cloud providers are added and remove the default value.
```

Comment:
Remove this. This is not needed. We will have task tracking this if new provider is added later on.